### ISSUE TITLE
cbMarkerAnnotate crashes on getting COSMIC file

### ISSUE THREAD

**Issue by matthewspeir** [maximilianh/cellBrowser#77]
When I run cbMarkerAnnotate with no arguments, I get the following 404 error:
```
$ cbMarkerAnnotate markers.tsv markers.annotated.tsv 
Traceback (most recent call last):
  File "/cluster/home/mspeir/miniconda3/bin/cbMarkerAnnotate", line 10, in <module>
    sys.exit(cbMarkerAnnotateCli())
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/geneinfo.py", line 305, in cbMarkerAnnotateCli
    entrezToCosmic = parseCosmic(options.cosmic)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/geneinfo.py", line 135, in parseCosmic
    for row in staticFileNextRow(inFname):
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/cellbrowser.py", line 232, in staticFileNextRow
    fh = openStaticFile(relPath)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/cellbrowser.py", line 226, in openStaticFile
    absPath = getStaticFile(relPath)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/cellbrowser.py", line 220, in getStaticFile
    downloadStaticFile(relPath, absPath)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/site-packages/cellbrowser/cellbrowser.py", line 203, in downloadStaticFile
    data = urlopen(remoteUrl).read()
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/cluster/home/mspeir/miniconda3/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```
Would an option to skip these errors and just show them as warnings be useful?

**Comment by maximilianh** [comment]
Ah, my testing was bad, I forgot to remove the cellbrowserData directory in
my homedirectory. Now that I removed I can reproduce this. It's because the
file has a space in the URL.

should be fixed now, new version 0.4.46 on pip.

>

**Comment by maximilianh** [comment]
Thanks!
